### PR TITLE
Updated with schema gan and added plots for paper

### PR DESCRIPTION
**Pull request by EleniSmyrniou**

### CODE REVIEW

File: train_model.py [hunk 1]
```diff
@@ -62,17 +63,17 @@ def main(nb_episodes: int, cpt_env: CPTEnvironment, agent: DQLAgent, training_da
...
+    actions = [50, 100, 150]  # actions in number of pixels
     output_folder = "results"
 
     cpt_env = CPTEnvironment(actions,
                              max_nb_cpts=50,
                              weight_reward_cpt=0.5,
                              image_width=512,
                              max_first_step=20,
-                             interpolation_method=InverseDistance(nb_points=6),
+                             interpolation_method=SchemaGANInterpolator("P:/schemagan/model_000036.h5"),
```
Comment (brunozc) [inline train_model.py]: I would put the model locally and add the relative path.  at least in the commit. the same applies for the training_data_folder on line 66

File: validate_model.py [hunk 2]
```diff
@@ -78,17 +78,16 @@ def evaluate_model(cpt_env: CPTEnvironment, agent: DQLAgent, validation_data_fol
 
 # Example usage
 if __name__ == "__main__":
-    validation_data_folder = "./data_fabian/validation"
-    actions = [10, 25, 50, 100, 150]  # actions in number of pixel
-    nb_cpts = 4
-    output_folder = "results/validation"
+    training_data_folder = "P:/schemagan/synthetic_database/512x32_20k/validation"
```
Comment (brunozc) [inline validate_model.py]: same comment as before

File: validate_model.py [hunk 3]
```diff
@@ -78,17 +78,16 @@ def evaluate_model(cpt_env: CPTEnvironment, agent: DQLAgent, validation_data_fol
...
+    actions = [50, 100, 150]  # actions in number of pixels
+    output_folder = "results/validation_1"
```
Comment (brunozc) [inline validate_model.py]:
```suggestion
    output_folder = "results/validation"
```

File: validate_model.py [hunk 4]
```diff
@@ -78,17 +78,16 @@ def evaluate_model(cpt_env: CPTEnvironment, agent: DQLAgent, validation_data_fol
...
 
     cpt_env = CPTEnvironment(actions,
                              max_nb_cpts=50,
                              weight_reward_cpt=0.5,
                              image_width=512,
                              max_first_step=20,
-                             interpolation_method=InverseDistance(nb_points=6)
+                             interpolation_method=SchemaGANInterpolator("D:/model_000036.h5")
```
Comment (brunozc) [inline validate_model.py]: same comment as before. relative path

File: CPTSuperLearn/interpolator.py [hunk 5]
```diff
@@ -24,6 +26,33 @@ def predict(self, prediction_points: List[float]):
         raise NotImplementedError("Predict method must be implemented")
 
 
+class SchemaGANInterpolator(InterpolatorAbc):
```
Comment (brunozc) [inline CPTSuperLearn/interpolator.py:29]: missing docstring in this class and in the functions of the class

File: CPTSuperLearn/interpolator.py [hunk 6]
```diff
@@ -24,6 +26,33 @@ def predict(self, prediction_points: List[float]):
         raise NotImplementedError("Predict method must be implemented")
 
 
+class SchemaGANInterpolator(InterpolatorAbc):
+    def __init__(self, model_path: str):
+        self.model_path = model_path
+        self.model = load_model(model_path)
+        self.size_x = 512
```
Comment (brunozc) [inline CPTSuperLearn/interpolator.py:39]: create a check to see if the size of the images being interpolated is the same as the size of the schemaGAN model. raise error if not

File: train_model.py (hunk 1 above)
Comment (EleniSmyrniou) [reply train_model.py]: ok

File: validate_model.py (hunk 2 above)
Comment (EleniSmyrniou) [reply validate_model.py]: ok

File: validate_model.py (hunk 4 above)
Comment (EleniSmyrniou) [reply validate_model.py]: ok

File: CPTSuperLearn/interpolator.py (hunk 5 above)
Comment (EleniSmyrniou) [reply CPTSuperLearn/interpolator.py:29]: added

File: CPTSuperLearn/interpolator.py (hunk 6 above)
Comment (EleniSmyrniou) [reply CPTSuperLearn/interpolator.py:39]: ok!

**Pull request merged**